enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,131 @@
+
+## Motivation
+
+### Goals
+As an administrator, I would like to create or reuse my own networks, subnets and routing schemes that I can deploy OpenShift to.
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a list of subnets that should be used for the cluster. Since there is an expectation that networking is being shared, the installer cannot modify the networking setup, but changes required to the shared resources like Tags that do not affect the behavior for other tenants of the network will be made.
+- The installer validates the assumptions about the networking setup
+- Non-networking infrastructure resources that are specific to the cluster and resources owned by the cluster will be created by the installer. So resources like security groups, roles, RHCOS boot images, and ignition storage objects.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+As an administrator of a cluster using provider networking, I have a pre-configured provider network, that cannot be manipulated by openstack tenants, and I want to install openshift on it.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+The user provides a list of subnets to the installer in the openstack platform section of the install-config.yaml.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    region: us-west-2
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+### Subnets
+- Subnets must all be a part of the same OpenStack cloud
+- Each subnet must be able to be tagged, and have the capacity for at least one tag. This allows us to add the `kubernetes.io/cluster/.*:` shared tag to identify the subnet as part of the cluster. The k8s cloud provider code in kube-controller-manager uses this tag to find the subnets for the cluster to create LoadBalancer Services.
+- The CIDR block for each one must be contained by Networking.MachineCIDR.
+- Subnets must be able to reach the node that is deploying the installer, otherwise it will fail to complete the bootstrapping and installation.
```

Comment:
hmm.. this is weird

why would the subnets themselves have connectivity to the host running the installer, we don't host anything on that host..

Reply:
hmm yeah, the wording here is not great. I'll change it to "nodes on user provided subnet must be able to reach the node running the installer."

Reply:
> nodes on user provided subnet must be able to reach the node running the installer.

I'm really concerned about this requirement. Can you provide details as to why this is necessary?

Reply:
Doesn't the installer query the api in order to verify that the control plane is up?

Reply:
> Doesn't the installer query the api in order to verify that the control plane is up?

a) wouldn't that be the host running the installer should have access to the subnet.

b) on AWS we need access to the LB, here since these are VIP maybe direct connectivity is required.

Reply:
Yeah, I think the way you worded things for comment (a) is clearer than what I have. And yes, we don't have an external load balancer, so they will have to be able to reach a port on the nodes subnet that is set aside for the API.

Reply:
Is the requirement that the any  worker or master vm reach the bootstrap vm  using ipv4. ? 
Does the host executing the "openshift-install create cluster" command need access to all the subnets (again ipv4) or just to a subnet connecting it to the boostrap vm?